plugins/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -142,6 +190,39 @@
 from ansible_collections.community.general.plugins.module_utils.gitlab import auth_argument_spec, gitlab_authentication
 
 
+def vars_to_variables(vars):
+    # transform old vars to new variables structure
+    variables = list()
+    for item in vars.keys():
+        if (isinstance(vars.get(item), str) or
+           isinstance(vars.get(item), int) or
+           isinstance(vars.get(item), float)):
+            variables.append(
+                {
+                    "name": item,
+                    "value": str(vars.get(item))
```

Comment:
The old code did not convert values to string. This probably causes a change in behavior of the module.

Reply:
Ah, then this PR fixes another bug

```yml
    - name: same vars, diff scope
      community.general.gitlab_project_variable:
        api_url: https://gitlab.com
        project: markuman/dotfiles
        purge: yes
        vars:
          int_value: 12
      register: bug

    - debug:
        var: bug is changed
       
    - name: same vars, diff scope
      community.general.gitlab_project_variable:
        api_url: https://gitlab.com
        project: markuman/dotfiles
        purge: yes
        vars:
          int_value: 12
    
    - debug:
        var: bug is changed
```

results in

```
TASK [Gathering Facts] ********************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [same vars, diff scope] **************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [debug] ******************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "bug is changed": true
}

TASK [same vars, diff scope] **************************************************************************************************************************************************************************************************
changed: [localhost]

TASK [debug] ******************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "bug is changed": true
}

PLAY RECAP ********************************************************************************************************************************************************************************************************************
localhost                  : ok=5    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

GitLab returns all ENV variables always as string. Even if you put an int or float value to the API.  

The PR version results in 

```

TASK [debug] ******************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "bug is changed": false
}

TASK [same vars, diff scope] **************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [debug] ******************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "bug is changed": false
}

PLAY RECAP ********************************************************************************************************************************************************************************************************************
localhost                  : ok=5    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

First task is also not changed because the value exists from the previous run.